processor/transformprocessor/internal/metrics/func_aggregate_on_attributes_metrics.go

```diff
@@ -0,0 +1,419 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type aggregateOnAttributesArguments struct {
+	Type       string
+	Attributes []string
```

Comment:
```suggestion
	Attributes Optional[[]string]
```
I thought about this a bit, and I think we should make this optional. If it's not included, we take metric attributes as-is. If it is passed, we use this list to determine the list of attributes that should remain after aggregation.

The reason for this is that we can use the `keep_matching_keys` and `delete_matching_keys` functions to allow users to get their attributes exactly how they want them before aggregation. Supporting the same level of functionality in this function would feel unnecessary.

I think keeping it as an optional attribute may have value for a few reasons:
1. This seems like it is likely a common use-case for metric aggregation, so this may help users cut down on statements.
2. We keep function signature parity with the metrics transform processor, though this isn't crucial.
3. In my opinion, lists are easier to read than the Regex equivalent of `one|two|three`.

What do you think @odubajDT @TylerHelmuth?

Reply:
we can definitely make it optional

Reply:
I believe it is optional in metricstransformprocessor so it should be optional here